### PR TITLE
Move web3.utils to web3._utils

### PR DESCRIPTION
**Pull request by dylanjw**
### What was wrong?

`web3.utils` contains code for internal use, and that could be made more clear.

### How was it fixed?

`web3.utils` was moved to `web3._utils`

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by pipermerriam** [comment]
> Can AttributeDict be in different module then? Any project that uses python type hints will be hurt.

:+1: would you be up for opening a pull request which moves it to a new `web3.datastructures` module?

**Comment by pipermerriam** [comment]
cc @carver regarding attrdict.  I see `eth-account` uses the `attrdict` library.  Should we converge usage to one or the other?  IIRC the `attrdict` library provided more functionality than we needed but there wasn't anything explicitely wrong with it.  I do see that we wrap it to make it immutable which is desirable.

**Comment by carver** [comment]
> Can AttributeDict be in different module then? Any project that uses python type hints will be hurt.

Yeah, this sounds like a good reason to wait to move `utils` until v5. In case no one noticed, I really hate breaking downstream dependencies in a minor release... :D

**Comment by carver** [comment]
>  I see eth-account uses the attrdict library. Should we converge usage to one or the other?

Yeah, I think a small wrapper around it could go in an eth-utils minor release :+1: and then both web3 and eth-account can use the same implementation.

**Comment by pipermerriam** [comment]
Maybe we can reliably alias it?  I'd be fine special casing this move and adding a deprecation warning to the `web3.utils` module.

**Comment by Exef** [comment]
@pipermerriam ok, I'll create PR with the data structures this weekend.

File: web3/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    module_testing,
+    normalizers,
+    request,
+    rpc_abi,
+    six,
+    threads,
+    toolz,
+    transactions,
+    validation,
+)
```
Comment (dylanjw) [inline web3/utils/__init__.py]: @carver @pipermerriam Is this the right way to alias `utils` -> `_utils`?

Comment (carver) [reply web3/utils/__init__.py]: I don't have a strong opinion on it. Seems okay to me.

**Pull request merged**